Spell check of Flutter docs

gspencergoog opened the pull request:
No code changes, just comments: I spell-checked all the comments in the repo.

gspencergoog committed:
Initial spell check changes

natebosch commented on dev/bots/test.dart:
```diff
@@ -32,7 +32,7 @@ final String pubCache = path.join(flutterRoot, '.pub-cache');
 
 /// The arguments to pass to `flutter test` (typically the local engine
-/// configuration) -- prefilled with the arguments passed to test.dart.
+/// configuration) -- pre-filled with the arguments passed to test.dart.
```
"prefill" is more common than "pre-fill"

https://books.google.com/ngrams/graph?content=prefill%2C+pre-fill&year_start=1800&year_end=2000&corpus=15&smoothing=3&share=&direct_url=t1%3B%2Cprefill%3B%2Cc0%3B.t1%3B%2Cpre%20-%20fill%3B%2Cc0#t1%3B%2Cprefill%3B%2Cc0%3B.t1%3B%2Cpre%20-%20fill%3B%2Cc0

natebosch commented on packages/flutter/lib/src/painting/binding.dart:
```diff
@@ -106,7 +106,7 @@ mixin PaintingBinding on BindingBase, ServicesBinding {
   ///
   /// Objects that show text and/or measure text (e.g. via [TextPainter] or
-  /// [Paragraph]) should listen to this and redraw/remeasure.
+  /// [Paragraph]) should listen to this and redraw/re-measure.
```
"remeasure" is more common than "re-measure"

natebosch commented on packages/flutter/lib/src/painting/text_painter.dart:
```diff
@@ -83,7 +83,7 @@ class PlaceholderDimensions {
 /// See [Text.textWidthBasis], for example.
 enum TextWidthBasis {
-  /// Multiline text will take up the full width given by the parent. For single
+  /// Multi-line text will take up the full width given by the parent. For single
```
"multiline" is more common than "multi-line"

natebosch commented on packages/flutter/lib/src/painting/text_painter.dart:
```diff
@@ -178,7 +178,7 @@ class TextPainter {
   ///
   /// The [InlineSpan] this provides is in the form of a tree that may contain
-  /// multiple instances of [TextSpan]s and [WidgetSpan]s. To obtain a plaintext
+  /// multiple instances of [TextSpan]s and [WidgetSpan]s. To obtain a plain text
```
"plaintext" is (slightly) more common than "plain text"

https://books.google.com/ngrams/graph?content=plaintext%2C+plain+text&year_start=1800&year_end=2000&corpus=15&smoothing=3&share=&direct_url=t1%3B%2Cplaintext%3B%2Cc0%3B.t1%3B%2Cplain%20text%3B%2Cc0

natebosch commented on packages/flutter/lib/src/rendering/sliver_multi_box_adaptor.dart:
```diff
@@ -228,7 +228,7 @@ abstract class RenderSliverMultiBoxAdaptor extends RenderSliver
   ///
   /// 1. Verify that the children index in childList is in ascending order.
-  /// 2. Verify that there is no dangling keepalive child as the result of [move].
+  /// 2. Verify that there is no dangling keep-alive child as the result of [move].
```
"keepalive" is more common than "keep-alive"

natebosch commented on packages/flutter/lib/src/services/text_input.dart:
```diff
@@ -418,7 +418,7 @@ class TextInputConfiguration {
   /// This flag only affects Android. On iOS, suggestions are tied directly to
   /// [autocorrect], so that suggestions are only shown when [autocorrect] is
-  /// true. On Android autocorrection and suggestion are controlled separately.
+  /// true. On Android auto-correction and suggestion are controlled separately.
```
"autocorrect" is more common than "auto-correction"

natebosch reviewed (commented):
I think this spell checker may be too zealous on compound words...

gspencergoog committed:
Convert some hyphenated-words into compoundwords

gspencergoog commented:
Hmm. I much prefer all of the hyphenated ones, but maybe that just makes me uncommon. The non-hyphenated versions all look like made up words to me.

I reverted all the hyphenations I added, and all other hyphenations of these words in all the comments, so that we're at least consistent.

I kept `plain text` because it feels more descriptive and less a term of art (and it's nearly as common).

Hixie commented on packages/flutter/lib/src/material/slider_theme.dart:
```diff
@@ -1500,7 +1500,7 @@ abstract class BaseSliderTrackShape {
 /// but is padded by the [RoundSliderOverlayShape] radius. The height is defined
 /// by the [SliderThemeData.trackHeight]. The color is determined by the
-/// [Slider]'s enabled state and the track segments's active state which are
+/// [Slider]'s enabled state and the track segments' active state which are
```
`segment's`?

gspencergoog replied:
Aaah, when I was doing my spell checking, I thought that the comment was referring to a plural number of segments' active state, not just one segment's active state.

Clearly not so.  Changed to `segment's`

Hixie commented on packages/flutter/lib/src/material/slider_theme.dart:
```diff
@@ -1706,7 +1706,7 @@ class RoundedRectSliderTrackShape extends SliderTrackShape with BaseSliderTrackS
 /// but is padded by the [RoundSliderOverlayShape] radius. The height is
 /// defined by the [SliderThemeData.trackHeight]. The color is determined by the
-/// [Slider]'s enabled state and the track segments's active state which are
+/// [Slider]'s enabled state and the track segments' active state which are
```
ditto

gspencergoog replied:
Fixed.

Hixie commented on packages/flutter_goldens/lib/flutter_goldens.dart:
```diff
@@ -156,7 +156,7 @@ abstract class FlutterGoldenFileComparator extends GoldenFileComparator {
   }
 
-  /// Prepends the golden Uri with the library name that encloses the current
+  /// Prepends the golden URI with the library name that encloses the current
```
URL

You should ban URI from your dictionary

gspencergoog replied:
OK, done.  And replaced all (reasonable) instances of URI with URL in the other places where they appeared.

Hixie commented on packages/flutter_tools/lib/src/asset.dart:
```diff
@@ -257,11 +257,11 @@ class _Asset {
   final Uri relativeUri;
 
-  /// A platform-independent Uri representing the entry for the asset manifest.
+  /// A platform-independent URI representing the entry for the asset manifest.
```
ditto

Hixie commented on packages/flutter_tools/lib/src/build_runner/web_fs.dart:
```diff
@@ -100,7 +100,7 @@ class WebFs {
   );
 
-  /// The server uri.
+  /// The server URI.
```
ditto

Hixie commented on packages/flutter_tools/lib/src/build_system/source.dart:
```diff
@@ -182,7 +182,7 @@ class SourceVisitor implements ResolvedFiles {
+  /// This source is a file URI which contains some references to magic
   /// environment variables.
   const factory Source.pattern(String pattern, { bool optional }) = _PatternSource;
   /// The source is provided by an [Artifact].
```
ditto x3

Hixie commented on packages/flutter_tools/lib/src/dart/package_map.dart:
```diff
@@ -46,7 +46,7 @@ class PackageMap {
   String pathForPackage(Uri packageUri) => uriForPackage(packageUri).path;
 
-  /// Returns the path to [packageUri] as Uri.
+  /// Returns the path to [packageUri] as URI.
```
ditto

Hixie commented on packages/flutter_tools/lib/src/devfs.dart:
```diff
@@ -534,5 +534,5 @@ class DevFS {
 }
 
-/// Converts a platform-specific file path to a platform-independent Uri path.
+/// Converts a platform-specific file path to a platform-independent URI path.
```
ditto

Hixie commented on packages/flutter_tools/lib/src/device.dart:
```diff
@@ -545,7 +545,7 @@ class DebuggingOptions {
   final String port;
   final String hostname;
-  /// A file where the vmservice uri should be written after the application is started.
+  /// A file where the vmservice URI should be written after the application is started.
```
ditto

Hixie commented:
![LGTM](http://junkyard.damowmow.com/577)

gspencergoog committed:
Ban URI from lexicon

dramosv90 commented on dev/devicelab/lib/framework/framework.dart:
```diff
@@ -143,7 +143,7 @@ class _TaskRunner {
   }
 
-  /// Disables the keep-alive port, allowing the VM to exit.
+  /// Disables the keepalive port, allowing the VM to exit.
```
```suggestion
  /// Disables the [keepAlivePort] allowing the VM to exit.
```

gspencergoog replied:
That would be great, but it's a private member: `_keepAlivePort`, and there isn't any description to link to.

dramosv90 commented on packages/flutter_tools/lib/src/protocol_discovery.dart:
```diff
@@ -63,20 +63,20 @@ class ProtocolDiscovery {
   ///
-  /// When a new observatory URI is available in [logReader],
-  /// the URIs are forwarded at most once every [throttleDuration].
+  /// When a new observatory UR: is available in [logReader],
```
```suggestion
  /// When a new observatory URL: is available in [logReader],
```

gspencergoog replied:
Whoops.  Thanks!

dramosv90 commented on packages/flutter_tools/lib/src/test/flutter_platform.dart:
```diff
@@ -293,7 +293,7 @@ class FlutterPlatform extends PlatformPlugin {
 
   /// The test compiler produces dill files for each test main.
```
dill or dll?

jonahwilliams replied:
dill is the serialization format for kernel, a Dart intermediate language used by the tooling
